projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -599,4 +626,44 @@ public function create_publicize_connection( $request ) {
 			__( 'Something went wrong while creating a connection.', 'jetpack-publicize-pkg' )
 		);
 	}
+
+	/**
+	 * Update the post with information about shares.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 */
+	public function update_post_shares( $request ) {
+		$request_body = $request->get_json_params();
+
+		$post_id   = $request->get_param( 'id' );
+		$post_meta = $request_body['meta'];
+		$post      = get_post( $post_id );
+
+		if ( $post && $post->post_status === 'publish' && isset( $post_meta[ self::SOCIAL_SHARES_POST_META_KEY ] ) ) {
+			update_post_meta( $post_id, self::SOCIAL_SHARES_POST_META_KEY, $post_meta[ self::SOCIAL_SHARES_POST_META_KEY ] );
+			return rest_ensure_response( new WP_REST_Response() );
+		}
+
+		return new WP_Error(
+			'rest_cannot_edit',
+			__( 'Failed to update the post meta', 'jetpack-publicize-pkg' ),
+			array( 'status' => 500 )
+		);
+	}
+
+	/**
+	 * Permissions callback.
+	 */
+	public function update_post_shares_permission_callback() {
+		if ( Rest_Authentication::is_signed_with_blog_token() ) {
```

Comment:
This function can simply return false if the user shouldn't be able to perform the operation.

```php
return Rest_Authentication::is_signed_with_blog_token();
```